packages/editor-preview/src/browser/editor-preview-widget.ts

```diff
@@ -0,0 +1,226 @@
+    }
+
+    replaceEditorWidget(editorWidget: EditorWidget) {
+        if (editorWidget === this.editorWidget_) {
+            return;
+        }
+        if (this.editorWidget_) {
+            this.editorWidget_.dispose();
+        }
+        this.editorWidget_ = editorWidget;
+        this.attachPreviewWidget(this.editorWidget_);
+        this.onResize(Widget.ResizeMessage.UnknownSize);
+    }
+
+    protected onActivateRequest(msg: Message) {
+        super.onActivateRequest(msg);
+        if (this.editorWidget_) {
+            this.editorWidget_.activate();
+        }
+    }
+
+    protected attachPreviewWidget(w: Widget) {
+        (this.layout as PanelLayout).addWidget(w);
+        this.title.label = w.title.label;
+        this.title.iconClass = w.title.iconClass;
+        this.title.caption = w.title.caption;
+
+        if (Saveable.isSource(w)) {
+            Saveable.apply(this);
+            const dirtyListener = w.saveable.onDirtyChanged(() => {
+                this.pinEditorWidget();
+            });
+            this.toDispose.push(dirtyListener);
+        }
+        w.parent = this;
+        w.activate();
+        this.onDidChangeTrackableWidgetsEmitter.fire([w]);
+    }
+
+    onAfterAttach(msg: Message) {
+        super.onAfterAttach(msg);
+        if (this.editorWidget_ && !this.editorWidget_.isAttached) {
+            this.attachPreviewWidget(this.editorWidget_);
+        }
+        this.addTabPinningLogic();
+    }
+
+    private lastParent: DockPanel;
+
+    protected addTabPinningLogic() {
+        const parent = this.parent;
+        if (!this.pinned_ && parent instanceof DockPanel) {
+            if (!this.lastParent) {
+                this.lastParent = parent;
+            }
+
+            const tabBar = find(parent.tabBars(), bar => bar.titles.indexOf(this.title) !== -1);
+
+            // Widget has been dragged into a different panel
+            if (this.lastParent !== parent || !tabBar) {
+                this.pinEditorWidget();
+                return;
+            }
+
+            const layoutListener = (panel: DockPanel) => {
+                if (tabBar !== find(panel.tabBars(), bar => bar.titles.indexOf(this.title) !== -1)) {
+                    this.pinEditorWidget();
+                }
+            };
+            parent.layoutModified.connect(layoutListener);
+            this.pinListeners.push({dispose: () => parent.layoutModified.disconnect(layoutListener)});
+
+            const tabMovedListener = (w: Widget, args: {title: Title<Widget>}) => {
+                if (args.title === this.title) {
+                    this.pinEditorWidget();
+                }
+            };
+            tabBar.tabMoved.connect(tabMovedListener);
+            this.pinListeners.push({dispose: () => tabBar.tabMoved.disconnect(tabMovedListener)});
+
+            requestAnimationFrame(() => {
+                const tabNode = tabBar.contentNode.children.item(tabBar.currentIndex);
```

Comment:
`tabNode` can possibly be null but no null checks are performed. The error fails the compilation and cli.